File: runner.py
import pygame
import sys
from pygame.locals import *
import time


COLOR1 = ((244, 194, 194))
COLOR2 = ((245, 245, 230))
BLACK = ((0,0,0))
pygame.init()
WHITE = (255,255,255)
# set up the window
size = (640, 640)
screen = pygame.display.set_mode(size)
pygame.display.set_caption("Chess Game")
pieces = []
global selected 
selected = False
# set up the board
board = pygame.Surface((640, 640))
board.fill(COLOR1)
    
# draw the board
def drawBoard():
    selected = False
    for x in range(0, 8, 2):
        for y in range(0, 8, 2):
            pygame.draw.rect(board, COLOR2, (x*80, y*80, 80, 80))
            pygame.draw.rect(board, COLOR2, ((x+1)*80, (y+1)*80, 80, 80))

# add the board to the screen
drawBoard()
screen.blit(board, (0, 0))

pygame.display.flip()

class Piece:
    def __init__(self, color, x, y, piece_type):
        self.color = color
        self.x = x
        self.y = y
        self.type = piece_type
        self.selected = False
        self.countMoves = 0

    def draw(self, letter,surface):
        PIECE = "assets/ChessAlpha2.ttf"
        mediumFont = pygame.font.Font(PIECE, 95)
        buttonRect = pygame.Rect(8,8,8,8)
        buttonText = mediumFont.render(letter, True, BLACK)
        buttonTextRect = buttonText.get_rect()
        buttonTextRect.center = buttonRect.center
        pygame.draw.rect(screen, WHITE, buttonRect)
        if self.type == "pawn":
            screen.blit(buttonText, (self.x*80+17,self.y*80))
        elif self.type == "rook":
            screen.blit(buttonText, (self.x*80+15,self.y*80))
        elif self.type == "king" or self.type == "queen":
            screen.blit(buttonText, (self.x*80+7,self.y*80))

        else:
            screen.blit(buttonText, (self.x*80+10,self.y*80))

    def getType(self):
        return self.type
    def getColor(self):
        return self.color
    def drawMoves(self):
        
        if(self.selected):
            drawBoard()
            screen.blit(board, (0, 0))
            drawPieces(pieces)
            self.selected = False
            

        else:
            drawBoard()
            screen.blit(board, (0, 0))
            drawPieces(pieces)
 
            pygame.draw.rect(screen, (244, 194, 80), (self.x*80, self.y*80, 80, 80))
            drawPieces(pieces)
            selected = True
            self.selected = True
     
        
        pygame.display.flip()
    
        
def drawPieces(pieces):
    for piece in pieces:
        if piece.getColor() == "white":
            if piece.getType() == "pawn":
                piece.draw("i",board)
            if piece.getType() == "rook":
                piece.draw("l",board)
            if piece.getType() == "bishop":
                piece.draw("j",board)
            if piece.getType() == "knight":
                piece.draw("k",board)
            if piece.getType() == "king":
                piece.draw("n",board)
            if piece.getType() == "queen":
                piece.draw("m",board)

        else:
            if piece.getType() == "pawn":
                piece.draw("I",board)
            if piece.getType() == "rook":
                piece.draw("L",board)
            if piece.getType() == "bishop":
                piece.draw("J",board)
            if piece.getType() == "knight":
                piece.draw("K",board)
            if piece.getType() == "king":
                piece.draw("N",board)
            if piece.getType() == "queen":
                piece.draw("M",board)
# set up the pieces

for i in range(8):
    pieces.append(Piece("black", i, 1, "pawn"))
    pieces.append(Piece("white", i, 6, "pawn"))

for i in range(1):
    pieces.append(Piece("black", 0, 0, "rook"))
    pieces.append(Piece("black", 2, 0, "bishop"))
    pieces.append(Piece("black", 1, 0, "knight"))
    pieces.append(Piece("white", 0, 7, "rook"))
    pieces.append(Piece("white", 2, 7, "bishop"))
    pieces.append(Piece("white", 1, 7, "knight"))

    pieces.append(Piece("black", 7, 0, "rook"))
    pieces.append(Piece("black", 5, 0, "bishop"))
    pieces.append(Piece("black", 6, 0, "knight"))
    pieces.append(Piece("white", 7, 7, "rook"))
    pieces.append(Piece("white", 5, 7, "bishop"))
    pieces.append(Piece("white", 6, 7, "knight"))
    pieces.append(Piece("black", 4, 0, "king"))
    pieces.append(Piece("black", 3, 0, "queen"))
    pieces.append(Piece("white", 4, 7, "king"))
    pieces.append(Piece("white", 3, 7, "queen"))

drawPieces(pieces)
# draw the pieces
pygame.display.flip()

dummyPiece = Piece(None,None,None,None)
lastSelected = dummyPiece

# main loop
while True:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            pygame.quit()
            sys.exit()


        if event.type == pygame.MOUSEBUTTONDOWN and selected:
            pos = pygame.mouse.get_pos()
            # convert the position to board coordinates
            x = (pos[0]) // 80
            y = (pos[1]) // 80

            invalidMove = False
            selectedPiece = dummyPiece

            
            for piece in pieces:
                if piece.x == x and piece.y == y and piece.color == lastSelected.color:
                    # if moving to where a piece is currently, stops it from moving
                    selected = False;
                    drawBoard()
                    screen.blit(board, (0, 0))
                    drawPieces(pieces)
                    invalidMove = True
                    piece.selected = False
                    lastSelected = piece
                    break
            if invalidMove:
                continue
            for piece in pieces:
                #deletes piece if same location as moved
                if piece.x == x and piece.y == y :
                    pieces.remove(piece)
                # moves if it is a valid location

                if (piece.selected == True):
                    piece.x = x 
                    piece.y = y 
                    print(piece.type)
                    piece.drawMoves()
                    lastSelected = piece
                    selected = False
                    piece.selected = False
                    piece.countMoves += 1
                    #time.sleep(0.4)
                    break
                
                piece.selected = False
                
            pygame.display.flip()
            time.sleep(0.2)

            
            
        elif event.type == pygame.MOUSEBUTTONDOWN and not selected:
            # get the position of the click
            pos = pygame.mouse.get_pos()
            # convert the position to board coordinates
            x = (pos[0]) // 80
            y = (pos[1]) // 80
            
            
            # find the piece at the clicked position
            for piece in pieces:
                if piece.x == x and piece.y == y:
                    # move the piece
                    pos = pygame.mouse.get_pos()
                    
                    piece.x = x
                    piece.y = y
                    print(piece.type)
                    piece.drawMoves()
                    lastSelected = piece
                    selected = True
                    pygame.display.flip()
                    time.sleep(0.2)
                    piece.countMoves += 1
                    break
        pygame.display.flip()


                    


                    



      

    

File: runner.cpp
#include <SFML/Graphics.hpp>
#include <SFML/Window.hpp>
#include <SFML/System.hpp>
#include <string.h>
using namespace std;
    class Piece {
        

    public:

        std::string color;
        int x;
        int y;
        int pieceType;
        bool selected;
        int countMove


        Piece(const std::string col, int xpos, int ypos, const std::int piece_type){
            color = (col); 
            x =(xpos); 
            y= (ypos); 
            pieceType = (piece_type); 
             selected = false; 
             countMoves = 0;
        }
    }
        }

        void draw(sf::Text& letter, sf::RenderWindow& surface) {
            sf::Font PIECE;
            PIECE.loadFromFile("assets/ChessAlpha2.ttf");
            letter.setFont(PIECE);
            letter.setCharacterSize(95);
            letter.setStyle(sf::Text::Bold);
            letter.setFillColor(BLACK);

            if (type == "pawn") {
                letter.setPosition(sf::Vector2f(x * 80 + 17, y * 80));
            }
            else if (type == "rook") {
                letter.setPosition(sf::Vector2f(x * 80 + 15, y * 80));
            }
            else if (type == "king" || type == "queen") {
                letter.setPosition(sf::Vector2f(x * 80 + 7, y * 80));
            }
            else {
                letter.setPosition(sf::Vector2f(x * 80 + 10, y * 80));
            }

            sf::RectangleShape buttonRect(sf::Vector2f(8, 8));
            buttonRect.setPosition(8, 8);
            buttonRect.setFillColor(sf::Color::White);
            surface.draw(buttonRect);
            surface.draw(letter);
        }

        std::string getType() const {
            return type;
        }

        std::string getColor() const {
            return color;
        }

        void drawMoves(sf::RenderWindow& window, sf::RectangleShape& board, std::vector<std::unique_ptr<Piece>>& pieces) {
            if (selected) {
                board.setFillColor(COLOR1);
                window.draw(board);
                for (auto& piece : pieces) {
                    sf::Text letter;
                    if (piece->getColor() == "white") {
                        if (piece->getType() == "pawn") {
                            letter.setString("i");
                        }
                       


int main()
{
    sf::Color COLOR1(244, 194, 194);
    sf::Color COLOR2(245, 245, 230);
    sf::Color BLACK(0, 0, 0);
    sf::RenderWindow window(sf::VideoMode(640, 640), "Chess Game");
    window.setFramerateLimit(60);
    std::vector<std::unique_ptr<Piece>> pieces;
    bool selected = false;

    // set up the board
    sf::RectangleShape board(sf::Vector2f(640, 640));
    board.setFillColor(COLOR1);

    // draw the board
    for (int x = 0; x < 8; x += 2) {
        for (int y = 0; y < 8; y += 2) {
            sf::RectangleShape square(sf::Vector2f(80, 80));
            square.setFillColor(COLOR2);
            square.setPosition(sf::Vector2f(x * 80, y * 80));
            board.draw(square);

            sf::RectangleShape square2(sf::Vector2f(80, 80));
            square2.setFillColor(COLOR2);
            square2.setPosition(sf::Vector2f((x + 1) * 80, (y + 1) * 80));
            board.draw(square2);
        }
    }

    // add the board to the screen
    window.draw(board);
